lib/tapctl.ml

```diff
@@ -112,12 +113,12 @@ module Dummy = struct
     Unixext.write_string_to_file filename str
 
   let find_next_unused_number list =
```

Comment:
I think this function could use a comment what it does. Is it delivering the first hole in a list? The first number greater than all existing numbers?

Reply:
Yeah, it's not easy to parse.

It returns the lowest (k + 1) where k is an in the list and (k + 1) is not in the list.

Reply:
To find the first of something using a fold, the result must use an option or approach the problem from the other end to find the last instance of the problem. Fold keeps iterating after the first instance is found.

Reply:
The set order is reversed for that reason (`IntRevSet`).

The discrepancy between the current implementation and the proposed one arises when the list is _not_ ordered from lowest to highest. When the list is ordered that way the behaviour is the same.

Reply:
Looking at the surrounding code, the list is used to simulate the behaviour of tapctl.

When an element is added to the list is always prepended to the list. Meaning it's in reverse order. I don't think there's an effort to keep the list in order.

I think this causes the previous code to _not_ use the lowest possible free spot, just the first one it can find while going though the list. This behaviour doesn't try to reuse the numbers that have been discarded.

I want to test the new behaviour and see if it causes problems. Otherwise I'd prefer to keep the new implementation just to try to reduce the highest number used on runtime.

Reply:
To find the lowest gap, isn't the easiest solution to sort the list and to traverse it from low to high?

Reply:
I've changed it to use a normal set, I hope now it's easier to understand.